website/project/views/register.py

```diff
@@ -395,14 +395,14 @@ def project_before_register(auth, node, **kwargs):
         if archive_errors:
             error = archive_errors()
             if error:
-                errors.append(error)
+                errors[addon.config.short_name] = error
                 continue
         name = addon.config.short_name
         if name in settings.ADDONS_ARCHIVABLE:
             messages[settings.ADDONS_ARCHIVABLE[name]]['addons'].add(addon.config.full_name)
         else:
             messages['none']['addons'].add(addon.config.full_name)
-    errors = [e for e in errors if e]
+    errors = errors.values()
```

Comment:
This could be easier to follow if you don't overwrite the `errors` variable. How about using a new variable like `error_messages`?